### PR TITLE
Convert issue list checkboxes to native

### PR DESCRIPTION
**Pull request by silverwind**
Use native instead of fomantic checkboxes in issue list. Benefits include no more JS pop-in on load and perfect a11y.

Before, with JS pop-in:

Screenshot 2023-03-20 at 17 02 02

After, Firefox on macOS:

Screenshot 2023-03-20 at 17 01 26

After, Chrome on macOS:

Screenshot 2023-03-20 at 17 01 42

I opted to not do styling yet but I see that the inconsistency between browsers may already be reason enough on doing it. I think if we style them, there should be one global style, including markdown ones which currently have custom styling.

### CODE REVIEW

File: web_src/js/features/common-issue.js [hunk 1]
```diff
@@ -58,16 +57,9 @@ export function initCommonIssue() {
...
       window.location.reload();
     });
   });
-
-  // NOTICE: This event trigger targets Firefox caching behaviour, as the checkboxes stay
-  // checked after reload trigger checked event, if checkboxes are checked on load
-  $('.issue-checkbox input[type="checkbox"]:checked').first().each((_, e) => {
-    e.checked = false;
-    $(e).trigger('click');
-  });
```
Comment (silverwind) [inline web_src/js/features/common-issue.js:72]: This seems no longer neccesary, and in fact it caused an issue where all checkboxes were being checked after page reload.

**Comment by silverwind** [comment]
As for styling, I think doing something like Firefox does, but with thinner border on unchecked state would be my preference.

**Comment by silverwind** [comment]
Some WIP on checkboxes: https://jsfiddle.net/silverwind/g58fr0zm/

**Comment by wxiaoguang** [comment]
> I opted to not do styling yet but I see that the inconsistency between browsers may already be reason enough on doing it. I think if we style them, there should be one global style, including markdown ones which currently have custom styling.

IMO I think the native style is good enough, then we do not need to fix these styles again and again, it saves everyone's time.

**Comment by delvh** [comment]
> Edit: marking as draft until styling is in.

I agree with @wxiaoguang, I don't think we need to make it overly complicated, I'm already happy to get rid of some Fomantic UI checkboxes

**Comment by silverwind** [comment]
Well, we can land this as-is. I plan to follow up with checkbox module removal later, and with that, I guess I would also remove the custom checkbox styles on markup.

Still, I do think I prefer styling these checkboxes eventually, the Chrome ones look especially bad currently 😉

**Review by delvh — APPROVED** [review]

File: web_src/js/features/common-issue.js [hunk 2]
```diff
@@ -58,16 +57,9 @@ export function initCommonIssue() {
       // checkboxes stay checked after reload
       if (action === 'close' || action === 'open') {
         // uncheck all checkboxes
-        $('.issue-checkbox input[type="checkbox"]').each((_, e) => { e.checked = false });
+        $('.issue-checkbox').each((_, e) => { e.checked = false });
```
Comment (silverwind) [inline web_src/js/features/common-issue.js]: It might be that this workaround is also no longer necessary, will verify.

Comment (silverwind) [reply web_src/js/features/common-issue.js]: Added `autocomplete=off` attribute to the checkboxes and removed these hacky JS blocks. Outcome is the same, no state caching between page reloads.

File: templates/repo/issue/list.tmpl [hunk 3]
```diff
@@ -29,9 +29,7 @@
 		<div id="issue-filters" class="ui stackable grid">
 			<div class="six wide column">
 				{{if $.CanWriteIssuesOrPulls}}
-					<div class="ui checkbox issue-checkbox-all gt-vm">
-						<input type="checkbox" title="{{.locale.Tr "repo.issues.action_check_all"}}">
-					</div>
+					<input type="checkbox" autocomplete="off" class="issue-checkbox-all gt-vm gt-mr-4" title="{{.locale.Tr "repo.issues.action_check_all"}}">
```
Comment (wxiaoguang) [inline templates/repo/issue/list.tmpl:32]: Does checkbox need `autocomplete="off"`?

----

I guess the "state caching between page reloads" are related to "Cache-Control" header? Is there any document about how autocomplete affects the "state caching"?

Comment (wxiaoguang) [reply templates/repo/issue/list.tmpl:32]: And, `title` vs tippy tooltip?

(and  below)

Comment (delvh) [reply templates/repo/issue/list.tmpl:32]: > Does checkbox need `autocomplete="off"`?

I guess see https://github.com/go-gitea/gitea/pull/23596#discussion_r1148405657?

Comment (wxiaoguang) [reply templates/repo/issue/list.tmpl:32]: I don't think it's the same thing. `autocomplete="off"` is designed for helping users fill forms, but not for playing with "state caching", see my updated comment.

Comment (silverwind) [reply templates/repo/issue/list.tmpl:32]: `autocomplete="off"` works and eliminates a batch of hacky JS. Also, GitHub does the same. Read https://stackoverflow.com/questions/299811 for details, it is the most accepted answer.

Comment (wxiaoguang) [reply templates/repo/issue/list.tmpl:32]: It looks unclear to me. I think it really needs some documents about: 

* Why there must be a "autocomplete=off" ? What it is used for?
* Is it related to Cache-Control? Does `no-store` resolve the problem? (ps: `no-store` couldn't be used due to #22604)
* Does it affect all browsers? Or only some browsers?
* When the developers should (must) use this attribute? And when not?
* Shold Gitea add "autocomplete=off" for all checkboxes?
* Could it be supported by framework level? eg: auto-add the attribute?

----

Without a clear document/comment, future contributors would make a mess again.

Comment (silverwind) [reply templates/repo/issue/list.tmpl:32]: I can add a comment, but to my knowledge this a "feature" specific to Firefox.

> And, title vs tippy tooltip?

I'd question whether to just remove the title as it servers no real purpose and just echoes the issue title and I don't like these `«` characters in use there.

Comment (wxiaoguang) [reply templates/repo/issue/list.tmpl:32]: But I guess you can't remove the `title` due to #22593 , maybe use `aria` attribute instead

**Comment by delvh** [comment]
ping for other maintainers.

**Comment by wxiaoguang** [comment]
> ping for other maintainers.

Hmm, also not for me, but the comment  https://github.com/go-gitea/gitea/pull/23596#discussion_r1148543320 is not resolved yet

**Comment by silverwind** [comment]
I'll do the title removal. `autocomplete` attribute will stay.

File: templates/repo/issue/list.tmpl (hunk 3 above)
Comment (silverwind) [reply templates/repo/issue/list.tmpl:32]: 1. I moved the `title` on individual items to `aria-label`, I think it's a good compromise to keep `title` on the "check all" box. Also, I replaced `«` with double quotes, which should read better in a screenreader.
2. I don't feel like explaining the checkbox autocomplete=off. Above stackoverflow link is the first google result for "checkbox autocomplete off" so any curious developer will find and understand it. We shouldn't need to litter our codebase with comments that explain easily googleable stuff.

**Comment by silverwind** [comment]
Done and rebased. Should be ready now.

**Review by lunny — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Comment by jolheiser** [comment]
🎺 🤖

**Pull request merged**